apps/oxlint/src/run.rs

```diff
@@ -40,12 +40,28 @@ pub type JsLintFileCb = ThreadsafeFunction<
         u32,                // Buffer ID
         Option<Uint8Array>, // Buffer (optional)
         Vec<u32>,           // Array of rule IDs
+        Vec<u32>,           // Array of options IDs
         String,             // Stringified settings effective for the file
     )>,
     // Return value
     String, // `Vec<LintFileResult>`, serialized to JSON
     // Arguments (repeated)
-    FnArgs<(String, u32, Option<Uint8Array>, Vec<u32>, String)>,
+    FnArgs<(String, u32, Option<Uint8Array>, Vec<u32>, Vec<u32>, String)>,
+    // Error status
+    Status,
+    // CalleeHandled
+    false,
+>;
+
+/// JS callback to setup configs.
+#[napi]
+pub type JsSetupConfigsCb = ThreadsafeFunction<
+    // Arguments
+    FnArgs<(String,)>, // Stringified options array
+    // Return value
+    String, // Result ("ok" or error message)
+    // Arguments (repeated)
+    FnArgs<(String,)>,
```

Comment:
Because there's only 1 argument here, can skip the `FnArgs` abstraction.

```suggestion
pub type JsSetupConfigsCb = ThreadsafeFunction<
    // Arguments
    String, // Options array as JSON
    // Return value
    String, // Result ("ok" or error message)
    // Arguments (repeated)
    String,
```

If that gives you problems (weird errors elsewhere due to the change), don't worry about it - we can change it in a follow-up. NAPI-RS's API is pretty hairy.